jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/util/JKubeProjectUtilTest.java

```diff
@@ -70,4 +77,72 @@ public void hasDependencyWithGroupIdWithNoDependency(@Mocked JavaProject project
     // Then
     assertThat(result).isFalse();
   }
+
+  @Test
+  public void getFinalOutputArtifact_withNothingProvided_returnsNull() {
+    // Given
+    JavaProject javaProject = JavaProject.builder().build();
+
+    // When
+    File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(javaProject);
+
+    // Then
+    assertThat(finalOutputArtifact).isNull();
+  }
+
+  @Test
+  public void getFinalOutputArtifact_withProjectArtifactVersion_returnsValidArtifact() throws IOException {
+    // Given
+    File artifactFile = new File(temporaryFolder.getRoot(), "foo-test-1.0.0.jar");
+    boolean artifactFileCreated = artifactFile.createNewFile();
+    JavaProject javaProject = JavaProject.builder()
+        .artifactId("foo-test")
+        .version("1.0.0")
+        .packaging("jar")
+        .buildDirectory(temporaryFolder.getRoot())
+        .build();
+
+    // When
+    File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(javaProject);
+
+    // Then
+    assertThat(artifactFileCreated).isTrue();
+    assertThat(finalOutputArtifact).hasName("foo-test-1.0.0.jar");
+  }
+
+  @Test
+  public void getFinalOutputArtifact_withBuildFinalName_returnsValidArtifact() throws IOException {
+    // Given
+    File artifactFile = new File(temporaryFolder.getRoot(), "foo-test-final.jar");
+    boolean artifactFileCreated = artifactFile.createNewFile();
+    JavaProject javaProject = JavaProject.builder()
+        .buildFinalName("foo-test-final")
+        .packaging("jar")
+        .buildDirectory(temporaryFolder.getRoot())
+        .build();
+
+    // When
+    File finalOutputArtifact = JKubeProjectUtil.getFinalOutputArtifact(javaProject);
+
+    // Then
+    assertThat(artifactFileCreated).isTrue();
+    assertThat(finalOutputArtifact).hasName("foo-test-final.jar");
+  }
+
+  @Test
+  public void getFinalOutputArtifact_withArtifact_returnsValidArtifact() throws IOException {
+    // Given
+    File artifactFile = new File(temporaryFolder.getRoot(), "foo-test-1.0.0.jar");
+    boolean artifactFileCreated = artifactFile.createNewFile();
+    JavaProject javaProject = JavaProject.builder()
+        .artifact(artifactFile)
+        .build();
```

Comment:
We can improve this test to showcase the fallback precedence. 

`getFinalOutputArtifact_withArtifactAndBuildFinalNameAndPackaging_returnsArtifact()`